File: app/src/main/java/com/divercity/app/features/profile/selectmajor/SelectMajorActivity.java
package com.divercity.app.features.profile.selectmajor;

import android.content.Context;
import android.content.Intent;

import com.divercity.app.core.base.BaseFragment;
import com.divercity.app.core.base.BaseFragmentActivity;

/**
 * Created by lucas on 17/10/2018.
 */

public class SelectMajorActivity extends BaseFragmentActivity {

    public static Intent getCallingIntent(Context context){
        return new Intent(context, SelectMajorActivity.class);
    }

    @Override
    protected BaseFragment fragment() {
        return SelectMajorFragment.newInstance();
    }
}

File: app/src/main/java/com/divercity/app/features/onboarding/selectmajor/SelectMajorActivity.kt
package com.divercity.app.features.onboarding.selectmajor

import android.content.Context
import android.content.Intent
import com.divercity.app.core.base.BaseActivity
import com.divercity.app.core.base.BaseFragment

/**
 * Created by lucas on 17/10/2018.
 */

class SelectMajorActivity : BaseActivity() {

    companion object {
        private const val INTENT_EXTRA_PARAM_PROGRESS = "progress"

        fun getCallingIntent(context: Context, progress : Int) : Intent {
            val intent = Intent(context, SelectMajorActivity::class.java)
            intent.putExtra(INTENT_EXTRA_PARAM_PROGRESS,progress)
            return intent
        }
    }

    override fun fragment(): BaseFragment = SelectMajorFragment.newInstance(
            intent.getIntExtra(INTENT_EXTRA_PARAM_PROGRESS, 0))
}

